pkg/cloud/google/machineactuator.go

```diff
@@ -796,19 +800,51 @@ func getOrNewComputeServiceForMachine(params MachineActuatorParams) (GCEClientCo
 	if params.ComputeService != nil {
 		return params.ComputeService, nil
 	}
-	// The default GCP client expects the environment variable
-	// GOOGLE_APPLICATION_CREDENTIALS to point to a file with service credentials.
-	client, err := google.DefaultClient(context.TODO(), compute.ComputeScope)
-	if err != nil {
-		return nil, err
+
+	var client *http.Client
+	var err error
+	// If specified in the GCE config, use the alternative authentication.
+	if params.CloudConfigPath != "" {
+		glog.Info("Trying to get open the GCE config")
+		client, err = clientWithAltTokenSource(params.CloudConfigPath)
 	}
+	if params.CloudConfigPath == "" || err != nil {
+		glog.Info("Using the default GCP client")
+		if err != nil {
+			glog.Warningf("Could not create an alternative auth client: %q", err)
+		}
+		// The default GCP client expects the environment variable
+		// GOOGLE_APPLICATION_CREDENTIALS to point to a file with service credentials.
+		client, err = google.DefaultClient(context.TODO(), compute.ComputeScope)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	computeService, err := clients.NewComputeService(client)
 	if err != nil {
 		return nil, err
 	}
 	return computeService, nil
 }
 
+func clientWithAltTokenSource(gceConfigPath string) (*http.Client, error) {
+	glog.Info("Trying to get the alt token")
+	gceConfig := struct {
+		Global struct {
+			ProjectID string `gcfg:"project-id"`
+			TokenURL  string `gcfg:"token-url"`
+			TokenBody string `gcfg:"token-body"`
+		}
+	}{}
+	if err := warnings.FatalOnly(gcfg.ReadFileInto(&gceConfig, gceConfigPath)); err != nil {
+		return nil, err
+	}
+	tokenSource := clients.NewAltTokenSource(gceConfig.Global.TokenURL, gceConfig.Global.TokenBody)
+	client := oauth2.NewClient(context.Background(), tokenSource)
```

Comment:
Would `oauth2.NoContext` be better than a background context? Or `context.TODO`?

In case it makes a difference, this week we decided that it would be a good idea to plumb context into the actuator interface, so it might makes sense soon to refactor this such that the http client gets created each time the actuator needs to call out to gcp so that we maintain the calling context down the call stack.

Reply:
I'd rather submit a change :-)

So if it's okay with you, I'll leave this as is so that "context.TODO" can be an indication for later refactoring for making sure that context is used and used appropriately.

Reply:
context.TODO sounds great. I didn't mean to imply we should change it now, but a TODO makes us remember that we should change it in the future.